minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/RunMiNiFi.java

```diff
@@ -1600,9 +1622,13 @@ private void restartInstance() throws IOException {
         }
     }
 
-    private static void performTransformation(InputStream configIs, String configDestinationPath) throws ConfigurationChangeException, IOException {
-        try {
-            ConfigTransformer.transformConfigFile(configIs, configDestinationPath);
+    private static ByteBuffer performTransformation(InputStream configIs, String configDestinationPath) throws ConfigurationChangeException, IOException {
+        try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+                TeeInputStream teeInputStream = new TeeInputStream(configIs, byteArrayOutputStream)) {
+
+            ConfigTransformer.transformConfigFile(teeInputStream, configDestinationPath);
+
+            return ByteBuffer.wrap(byteArrayOutputStream.getUnderlyingBuffer());
```

Comment:
@JPercivall I think this is what was causing my constant restarting to happen.  If the underlying buffer isn't full, it will result in a bunch of null values at the end in the transformed config.